TypeCobol/Compiler/Diagnostics/Cobol85Checkers.cs

```diff
@@ -17,6 +17,12 @@ namespace TypeCobol.Compiler.Diagnostics {
 
     public class Cobol85CompleteASTChecker : AbstractAstVisitor
     {
+        public Cobol85CompleteASTChecker(List<Diagnostic> diagnostics)
```

Comment:
Why add this property ?
A Checker doesn't need to know about diagnostics. They are attached to Token, Line, CodeElement, Node, ...
As I understand it, you use this property so ProgramClassParserStep can retrieve it and then these diagnostics are merged with diagnostics of ProgramClassDocument.

I think a proper solution (but perhaps slower) would be to read all Diagnostics on Node from the CompilationUnit.AllDiagnostics() method.


There will be less code in this Checker and the possible others Checkers.